library/test/src/lib.rs

```diff
@@ -514,6 +516,18 @@ pub fn filter_tests(opts: &TestOpts, tests: Vec<TestDescAndFn>) -> Vec<TestDescA
     filtered
 }
 
+/// Custom handling of cases where there is only 1 test to be executed.
+/// This is needed because in such cases we may want to show more detailed information.
+/// For example for investigation purposes we may want to show the reasons why the test case was ignored.
+fn check_for_single_test_run(filtered_tests: &Vec<TestDescAndFn>) {
+    if filtered_tests.len() == 1 {
+        let test_desc = &filtered_tests.get(0).unwrap().desc;
+        if let Some(im) = &test_desc.ignore_message {
+            println!("test: {}, ignore message: {}", test_desc.name, im);
```

Comment:
I believe libtest always tries to put output into the formatters, so that we can support JSON and other ways of formatting. (At the very least grepping for `println` seems to not turn up most of libtest's current output).

So I think we'll want to adjust this to be compatible with the current way of doing things -- you can try to work back from library/test/src/formatters/mod.rs and seraching for writeln! perhaps? (I'm not super familiar with the details here, but feel free to ask for help if needed).